home/home_3.1/unittest.swift

```diff
@@ -0,0 +1,65 @@
+import XCTest
+
+func calculateValue(x: Double, y: Double, z: Double) -> Double {
+    let product = x * y * z
+
+    if product != 0 {
+        return 1 / product
+    } else {
+        let sum = x + y + z
+
+        if sum != 0 {
+            return 1 / sum
+        } else {
+            return x + (y + 1) * (z - 1)
+        }
+    }
+}
+
+class CalculateValueTests: XCTestCase {
+
+    func testNonZeroProduct() {
+        let x = 2.0
+        let y = 3.0
+        let z = 4.0
+        let expectedResult = 1 / (x * y * z)
```

Comment:
adding empty lines between logical blocks will make your unit tests more readable. For example all of your team may choose some style and therefore everyone from the team will be able to get the important parts form unit test automatically just baecause of those empty lines.

Reply:
also it is better to just calculate the value of this test case and place it here. You have the certain input set and you are expecting to get the certain numeric result even if the formula under the hood will be changed. 
Unit test should not duplicate formulas from implementation or even parts from it. Just inputs and expected output.
so let's calculate the values and place them across unit tests in this file.